Sample_Files/src/Context/LoginContext.php

```diff
@@ -79,60 +48,63 @@ private function pressLoginButton() {
    * @param string $username
    */
   public function iEnterAUsername($username) {
-    self::fillUsernameField($username);
+    $this->helperContext->getSession()
+      ->getPage()
+      ->fillField(self::FIELD_USERNAME, $username);
   }
   /**
    * @Given I enter the password :password
    *
    * @param string $password
    */
   public function iEnterAPassword($password) {
-    self::fillPasswordField($password);
+    $this->helperContext->getSession()
+      ->getPage()
+      ->fillField(self::FIELD_PASSWORD, $password);
   }
 
   /**
    * @Given I enter the username :username and password :password
    *
-   * @param string $username , $password
+   * @param string $username
+   * @param string $password
    */
   public function iEnterAUsernameAndPassword($username, $password) {
-    self::fillUsernameField($username);
-    self::fillPasswordField($password);
+    $this->helperContext->getSession()
+      ->getPage()
+      ->fillField(self::FIELD_USERNAME, $username);
+    $this->helperContext->getSession()
+      ->getPage()
+      ->fillField(self::FIELD_PASSWORD, $password);
   }
 
-
   /**
    * @Given I press login
    */
   public function iPressTheLoginButton() {
-    self::pressLoginButton();
+    $this->helperContext->getSession()
+      ->getPage()
+      ->pressButton(self::BUTTON_LOGIN);
   }
 
   /**
    * @Given I am still on the Login page
    */
   public function iAmStillOnTheLoginPage() {
     $current_url = $this->helperContext->getSession()->getCurrentUrl();
-    if (strpos($current_url, $this->loginPage->getPath()) === FALSE) {
+    if (strpos($current_url, $this->path) === FALSE) {
       throw new CWContextException("No longer on the Loginpage, but on {$current_url}.");
     }
   }
 
   /**
-   * @Given I should see the login failure message
-   */
-  public function iShouldSeeTheLoginFailureMessage() {
-    $this->helperContext->iCanSeeInTheRegion('Unrecognized username or password.', $this->loginPage->getMessageRegion(self::REGION_LOGIN_FAILURE));
-    $this->helperContext->iCanSeeInTheRegion('Have you forgotten your password?', $this->loginPage->getMessageRegion(self::REGION_LOGIN_FAILURE));
-  }
-
-  /**
    * @Given I verify the structure of the Login page
    */
   public function iVerifyTheStructureOfTheLoginPage() {
-    self::verifyFields($this->loginPage->getAllFields());
-    self::verifyButtons($this->loginPage->getAllButtons());
-    self::verifyRegions($this->loginPage->getAllRegions());
+    $this->verifyField(self::FIELD_USERNAME);
+    $this->verifyField(self::FIELD_PASSWORD);
+    $this->verifyButton(self::BUTTON_LOGIN);
+    $this->helperContext->iCanSeeInTheRegion('Enter your Cameron and Wilding username.', self::REGION_USERNAME_DESCRIPTION);
```

Comment:
C&W? Is this legit and flexible? Isn't it tying the code to our brand instead of keeping it neutral?